lib/spack/spack/cmd/patch.py

```diff
@@ -16,7 +16,8 @@
 
 
 def setup_parser(subparser):
-    arguments.add_common_arguments(subparser, ['no_checksum', 'specs'])
+    arguments.add_common_arguments(
+        subparser, ['no_checksum', 'deprecated', 'specs'])
```

Comment:
I'll again fold this into another issue, but grouping common arguments into "subsystems" is a really neat way that pants enables the configurability seen at https://v1.pantsbuild.org/options.html. Nothing to fix here, this is neat to learn.

Reply:
I would actually love for Spack to only use common arguments so that all of our command flags use identical short flags and help descriptions.